embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpanImpl.kt

```diff
@@ -14,11 +14,14 @@ import java.util.concurrent.atomic.AtomicReference
 internal class EmbraceSpanImpl(
     private val spanBuilder: SpanBuilder,
     override val parent: EmbraceSpan? = null,
-    private val spansRepository: SpansRepository? = null
+    private val spansRepository: SpansRepository? = null,
+    sessionSpan: Boolean = false
```

Comment:
This is fine for now, but I'm wondering if we should model this more explicitly. It's probably OK for now but I don't want to abuse this pattern if there are other specialized spans in the future that we want to wrap similarly.